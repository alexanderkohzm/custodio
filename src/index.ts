import express, { Express } from 'express'
import dotenv from 'dotenv'
import { router } from './routers/router'
import { connectToMongoDb } from './bootstrap/database';

dotenv.config();

const app = express();
const port = process.env.PORT

const startServer = async (app: Express, port: string) => {

  const mongoDbURL = process.env.MONGODB_URL!
  connectToMongoDb(mongoDbURL)

  app.use('/', router)
  app.listen(port, () => {
    console.log(`ðŸš€ Server is running at http://localhost:${port}`)
  })
}

startServer(app, port!);

export { app }