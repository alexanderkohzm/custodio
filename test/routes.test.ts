
import supertest from 'supertest'
import { app } from '../src/index'


describe(`routes test`, () => {

  const request = supertest

  it(`Should return an array of employees`, async () => {
    const response = await request(app).get('/employees');

    expect(response.status).toBe(200)

    const arrayOfEmployees = response._body

    // expect the salary field to be defined. 
    expect(arrayOfEmployees[0].salary).toBeDefined()
    // note: departmentName, divisionId, and divisionName are nullable

    // alternatively, just look at snapshots
    expect(arrayOfEmployees).toMatchSnapshot()
  })
})
