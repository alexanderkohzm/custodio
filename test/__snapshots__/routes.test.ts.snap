// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes test Should return an array of employees 1`] = `
[
  {
    "age": 25,
    "departmentId": 1,
    "departmentName": "r&d",
    "divisionId": 2,
    "divisionName": "",
    "hod": "ken",
    "jobGrade": "1",
    "joinedDate": "2022-04-05T00:00:00.000Z",
    "name": "Thomas Edison",
    "salary": 1000,
  },
  {
    "age": 52,
    "departmentId": null,
    "departmentName": "",
    "divisionId": 2,
    "divisionName": "tech",
    "hod": "alice",
    "jobGrade": "2",
    "joinedDate": "2022-04-05T00:00:00.000Z",
    "name": "Nikola Tesla",
    "salary": 2000,
  },
  {
    "age": 15,
    "departmentId": 1,
    "departmentName": "r&d",
    "divisionId": 2,
    "divisionName": "",
    "hod": "ken",
    "jobGrade": "3",
    "joinedDate": "2022-04-05T00:00:00.000Z",
    "name": "Michael Jackson",
    "salary": 3000,
  },
]
`;
